gnosis/safe-react
# Update route + move network name to slug

## What it solves
Resolves #2678

## How this PR fixes it
- New routes have been added.
- `HashRouter` has been exchanged with `BrowserRouter` in order to remove `#` from the URL.
- A new `/:network` slug has been added to routes.
- `:networkName` slug is appended with `window.location.hash`, replacing the current history stack entry.
- Bonus: router and history have been completely separated from Redux. `connect-react-router` was uninstalled and all selectors/history references updated accordingly.

## How to test it
- Enter old routes in the address bar and ensure that they are redirected to the new one.
- All links should go to the new routes.

## Notes
- We need to modify the deploy process when this successfully makes it out of QA because one package now serves all networks.
